src/std/misc/list.ss

```diff
@@ -40,6 +40,21 @@
 (defalias unique delete-duplicates)
 (defalias unique! delete-duplicates!)
 
+;; duplicates returns a cons cells (item . count) for every element
+;; that occurs more than once in the list.
+;;
+;; Example:
+;;  (duplicates ['a 1 'a]) => ((a . 2))
+(def (duplicates list (test equal?))
+  (def (proc v)
+    (let loop ((rest list) (count 0))
+      (if (pair? rest)
+        (loop (cdr rest) (if (test v (car rest)) (1+ count) count))
+        (if (> count 1) (cons v count) #f))))
```

Comment:
It may make sense to use a hash-table to keep occurrence counts for all elements in the list in a single pass; this is `O(n^2)`